src/main/java/com/netflix/imflibrary/IMFConstraints.java

```diff
@@ -254,10 +253,10 @@ else if (!audioChannelLabelSubDescriptor.getSoundfieldGroupLinkId()
                                 imfErrorLogger.addError(IMFErrorLogger.IMFErrors.ErrorCodes.IMF_CORE_CONSTRAINTS_ERROR, IMFErrorLogger.IMFErrors.ErrorLevels.NON_FATAL, IMF_ESSENCE_EXCEPTION_PREFIX +
                                         String.format("WaveAudioEssenceDescriptor in the IMFTrackFile represented by ID %s seems to indicate an Audio Bit Depth = %d, only 24 is allowed.", packageID.toString(), waveAudioEssenceDescriptor.getQuantizationBits()));
                             }
-
-                        } else {//Section 5.3.4.1 st2067-2:2016
-                            imfErrorLogger.addError(IMFErrorLogger.IMFErrors.ErrorCodes.IMF_CORE_CONSTRAINTS_ERROR, IMFErrorLogger.IMFErrors.ErrorLevels.FATAL, IMFConstraints.IMF_ESSENCE_EXCEPTION_PREFIX + String.format("Header Partition does not have a WaveAudioEssenceDescriptor set in the IMFTrackFile represented by ID %s", packageID.toString()));
                         }
+//                        else {//Section 5.3.4.1 st2067-2:2016
```

Comment:
Is it possible to restructure this logic so we can keep the 5.3.4.1 validation? 
At this point in the checks, do we have an easy way to distinguish between regular audio essence and IAB essence?

Reply:
There is no easy way to determine that a sound essence is an audio essence. Actually, I think the presence of the descriptor indicates that.